webapp/packages/core-authentication/src/UserInfoResource.ts

```diff
@@ -126,15 +130,34 @@ export class UserInfoResource extends CachedDataResource<UserInfo | null, void,
     });
 
     if (authInfo.userTokens && authInfo.authStatus === AuthStatus.Success) {
-      this.resetIncludes();
-      this.setData(await this.loader());
-      this.sessionResource.markOutdated();
+      await this.syncData();
     }
 
     return authInfo as AuthInfo;
   }
 
-  finishFederatedAuthentication(authId: string, linkUser?: boolean): ITask<UserInfo | null> {
+  async asyncAuthLogin(provider: string, { configurationId, linkUser, forceSessionsLogout }: IAsyncLoginOptions): Promise<AsyncAuthInfo> {
```

Comment:
Can we use this function for any authentication type? Or just for federated? If we can't use it for any authentication type, I suggest changing its name to reflect what kind of authentication type it is for

Reply:
lets call it federatedLogin